### DISCUSSION TITLE
Intended behavior of Field in Parent-Child-Inheritance

### DISCUSSION THREAD

**Discussion by florianfischer91** [pydantic/pydantic#4896] [Pydantic V2]
Given the following code
```python
class Parent(BaseModel):
    a: int = 3
    # a: Annotated[int, Field(3)]

class Child(Parent):
    b: str = Field(alias='bbb')
```
the resulting field infos of `Parent` and `Child` are
```bash
{'a': FieldInfo(annotation=int, required=False, default=3)} # Parent
{'a': FieldInfo(annotation=int, required=True), 'b': FieldInfo(annotation=str, required=True, alias='bbb', alias_priority=2)} # Child
```

You can see that for `Child` the inherited field `a` has no default value and is required. Using `Annotated` and `Field` (see comment) leads to the (in my opinion) correct field infos
```bash
{'a': FieldInfo(annotation=int, required=False, default=3)}
{'a': FieldInfo(annotation=int, required=False, default=3), 'b': FieldInfo(annotation=str, required=True, alias='bbb', alias_priority=2)}
```

This behavior applies to every case where a 'value' is assigned to a field. I know that V2 is still in development but i just stumpled over this and want to now whether this is intended and one should just use `Annotated` or still has to be implemented?

**Comment by mkdir700** [comment]
Hello, guys! I want to ask you how to get this Field infos. I get the following results by accessing the `__fileds__` property.

```python
{'a': ModelField(name='a', type=int, required=False, default=3)}
{'a': ModelField(name='a', type=int, required=False, default=3), 'b': ModelField(name='b', type=str, required=True, alias='bbb')}
```

**Reply by samuelcolvin** [reply]
so what's the question?

**Reply by mkdir700** [reply]
> so what's the question?

I'm not sure, so I want to know how he got the information.

This is different from what I got.

```python
{'a': FieldInfo(annotation=int, required=False, default=3)}
{'a': FieldInfo(annotation=int, required=False, default=3), 'b': FieldInfo(annotation=str, required=True, alias='bbb', alias_priority=2)}
```

**Reply by florianfischer91** [reply]
Are you on the main branch?

**Reply by mkdir700** [reply]
Yes, I'm using v1.10.4

**Reply by florianfischer91** [reply]
Well, my question is targeting V2 which is in the main branch (see pydantic/version.py) and still in development. If you are using 1.10.4 then you have installed pydantic either via pip or you are using the 1.10.x-branch

**Reply by mkdir700** [reply]
fine, thanks.😄

**Comment by samuelcolvin** [comment]
@florianfischer91 looks like a bug in V2, thanks for finding this.

The default should definitely still be there on the child, the following 3 fields should all be exactly the same

```py
class MyModel(BaseModel):
    a: int = 3
    a: Annotated[int, Field(3)]
    a: int = Field(3)
```

**Reply by samuelcolvin** [reply]
Discuss and fix on #4909
